ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueConsumption.scala

```diff
@@ -173,11 +179,24 @@ object KeyValueConsumption {
       case DamlLogEntry.PayloadCase.TRANSACTION_REJECTION_ENTRY =>
         transactionRejectionEntryToUpdate(recordTime, entry.getTransactionRejectionEntry)
 
+      case DamlLogEntry.PayloadCase.OUT_OF_TIME_BOUNDS_ENTRY =>
+        outOfTimeBoundsEntryToUpdate(recordTime, entry.getOutOfTimeBoundsEntry).toList
+
       case DamlLogEntry.PayloadCase.PAYLOAD_NOT_SET =>
         throw Err.InternalError("logEntryToUpdate: PAYLOAD_NOT_SET!")
     }
   }
 
+  private def resolveRecordTimeOrThrow(
+      recordTimeForUpdate: Option[Timestamp],
+      recordTimeFromLogEntry: Option[Timestamp]): Timestamp =
+    (recordTimeForUpdate, recordTimeFromLogEntry) match {
```

Comment:
Up for discussion whether we want to override the record time in the log entry with the record time for the update. My preference would be to have the record time from the log entry take precedence over the record time for the update. This way, we could selectively set the record time to the pre-execution time in some rejection log entries. (It's not really feasible in Concord, but we could do this in DAML-on-SQL if we wanted to; not sure that it would buy us anything in the end.)

Reply:
As discussed, changed it to always take the record time from the log entry whenever available.